openmc/tallies.py

```diff
@@ -1487,10 +1487,16 @@ def _outer_product(self, other, binary_op):
         other_filters = set(other.filters)
         filter_intersect = self_filters.intersection(other_filters)
 
-        # Align the shared filters to follow in each tally operand
+        # Align the shared filters in successive order
         for i, filter in enumerate(filter_intersect):
-            self_index = self.filters.index(filter)
-            other_filter = other.filters[self_index]
+            self_filter = self.filters[i]
+            other_filter = other.filters[i]
+
+            # If necessary, swap self filter
+            if self_filter != filter:
+                self = self.swap_filters(filter, self_filter)
```

Comment:
I just wanted to swap the filters, but since `swap_filters` returns a tally set `self` to that returned tally. Do you know if I can just remove `self =` and have the filter swap reflected in self?